game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -155,140 +51,141 @@ public void ping() {}
 
   public Chat(
       final Messengers messengers, final String chatName, final ChatSoundProfile chatSoundProfile) {
+    this.localPlayerName = messengers.getLocalNode().getPlayerName();
+    this.serverPlayerName = messengers.getServerNode().getPlayerName();
+    chatTransmitter = new JavaSocketChatTransmitter(this, chatName, messengers);
     this.chatSoundProfile = chatSoundProfile;
-    this.messengers = messengers;
-    statusManager = new StatusManager(messengers);
-    chatChannelName = ChatController.getChatChannelName(chatName);
-    this.chatName = chatName;
-    sentMessages = new SentMessagesHistory();
-
-    /*
-     * The order of events is significant.
-     *
-     * 1. Register our channel listener. Once the channel is registered, we are guaranteed that
-     * when we receive the response from our init message, our channel subscriber has been added,
-     * and will see any messages broadcasted by the server.
-     *
-     * 2. Call the init message on the server remote. Any add or join messages sent from the server
-     * will wait until we receive the init return value.
-     *
-     * 3. When we receive the init message response, initialize our state and release the latch
-     * so all pending messages will get processed. Messages may be ignored if the server
-     * version is incorrect. This all seems a lot more involved than it needs to be.
-     */
-    final IChatController controller = messengers.getRemoteChatController(chatName);
```

Comment:
So you did actually modify the class with this weird constraints.
You have my full respect ^^

Reply:
Thanks, I think one big thing I learned is "getRemote" returns a server endpoint and calls it. "getBroadcaster" invokes all client endpoints via the server.

The code behind this was tricky, I spent a few hours last night debugging a remote not found exception, it was kinda painful. On plus side, it all should now be a bit more hidden behind the transmitter.